### DISCUSSION TITLE
no-param-reassign error is set for @reduxjs/toolkit createSlice/createReducer

### DISCUSSION THREAD

**Discussion by OldMeM** [eslint/eslint#14759] [Q&A] [answered]
[@reduxjs/toolkit documentation ](https://redux.js.org/tutorials/fundamentals/part-8-modern-redux#immutable-updates-with-immer) suggests the below code for using `createSlice`

```
const todosSlice = createSlice({
  name: 'todos',
  initialState,
  reducers: {
    todoAdded(state, action) {
      const todo = action.payload
      state.entities[todo.id] = todo  <-- This line runs afoul of no-param-reassign
    },
  }
})
```

With a caveat : 
> You can only write "mutating" logic in Redux Toolkit's createSlice and createReducer because they use Immer inside! If you write mutating logic in reducers without Immer, it will mutate the state and cause bugs!

Is it possible to change eslint config for `no-param-reassign` rule to ignore the any code within `createSlice` or `createReducer` only?

**Comment by OldMeM** [comment]
For the time being I have disabled the rule using block comments

```
const todosSlice = createSlice({
  /* eslint-disable no-param-reassign */
  name: 'todos',
  initialState,
  reducers: {
    todoAdded(state, action) {
      const todo = action.payload
      state.entities[todo.id] = todo
    },
  }
  /* eslint-enable no-param-reassign */
})
```

I was hoping it could be something in built to avoid having to do this in every createSlice/createReducer function.

**Comment by nzakas** [comment]
Just FYI, this isn’t something the ESLint team deals with. You should contact the owner of the plugin for help with it.

**Comment by OldMeM** [comment, marked as answer]
I found an answer to this on futher searching. 
1. reduxjs/redux-toolkit#521
2. [Linting State Mutations](https://redux-toolkit.js.org/usage/immer-reducers#linting-state-mutations)

I should have searched for this first in their repo before coming over to eslint. :woman_facepalming:
